include/inexor/vulkan-renderer/world/cube.hpp

```diff
@@ -183,4 +183,6 @@ class Cube : public std::enable_shared_from_this<Cube> {
     [[nodiscard]] std::shared_ptr<Cube> neighbor(NeighborAxis axis, NeighborDirection direction);
 };
 
+std::shared_ptr<world::Cube> create_random_world(int max_depth, const glm::vec3 &position, unsigned int seed = 0);
```

Comment:
Why no fixed width integer types?

Reply:
Yeah `seed` should probably be `uint32_t`. Probably the same for `max_depth` too @IceflowRE?

Reply:
`max_depth` yes sure, `int` is only because of historical reasons here :D can both be changed.
Another note: You should document that seed `0` is not usable and results into a random seed.

Reply:
Right, I always forget to update the docs. Although thinking about it again, maybe the function should be:

```c++
std::shared_ptr<world::Cube> create_random_world(std::uint32_t max_depth, const glm::vec3 &position, std::optional<std::uint32_t> = std::nullopt);
```
so that `0` as a seed is still usable?

Reply:
I've made the seed an `std::optional` so 0 can still be used.